packages/vscode-extension/src/debug/npmLogHandler.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { spawn } from "child_process";
+import * as fs from "fs-extra";
+import * as path from "path";
+
+function getNpmCachePath(): Promise<string> {
+  const command = spawn("npm config get cache", {
+    shell: true,
+  });
+  return new Promise((resolve, reject) => {
+    command.stdout.on("data", (data) => {
+      resolve(`${data.toString().trim()}`);
+    })
+  });
+}
+
+async function getLatestNpmLogFile(npmLogPath: string): Promise<string | undefined> {
+  const files = await fs.readdir(npmLogPath);
+  if (files.length === 0) {
+      return undefined;
+  }
+  const latestNpmLogFile = files.reduce((previous, current, index, array) => {
+      return previous > current ? previous : current;
+  });
+  return path.join(npmLogPath, latestNpmLogFile);
+}
+
+export async function getNpmInstallErrorLog(cwd: string): Promise<Array<string> | undefined> {
+  const npmCachePath = await getNpmCachePath();
+  if (!fs.pathExists(npmCachePath)) {
+    return undefined;
+  }
+  const latestNpmLogFile = await getLatestNpmLogFile(path.join(npmCachePath, "_logs"));
+  if (latestNpmLogFile === undefined) {
+      return undefined;
+  }
+  const log = (await fs.readFile(latestNpmLogFile)).toString();
+  const cwdPattern = /\d+ verbose cwd (.*)/;
+  const cwdResult = log.match(cwdPattern);
+  // TODO: handle case sensitive path
+  if (!cwdResult || cwdResult[1].trim().toLowerCase() !== cwd.toLowerCase()) {
+      return undefined;
+  }
+  const errorPattern = /\d+ error .*/g;
```

Comment:
Add following info as well:
- `/\d+ verbose node .*/g` : to check the node version
- `/\d+ verbose npm .*/g` : to check the npm version
- `/\d+ verbose exit .*/g` : to double check the exit code

Reply:
Added.